exporter/containerimage/writer.go

```diff
@@ -380,11 +399,17 @@ func rewriteImageLayerWithEpoch(ctx context.Context, cs content.Store, desc ocis
 	return converterFn(ctx, cs, desc)
 }
 
-func (ic *ImageWriter) rewriteRemoteWithEpoch(ctx context.Context, opts *ImageCommitOpts, remote *solver.Remote) (*solver.Remote, error) {
+func (ic *ImageWriter) rewriteRemoteWithEpoch(ctx context.Context, opts *ImageCommitOpts, remote *solver.Remote, srcImg *dockerspec.DockerOCIImage) (*solver.Remote, error) {
 	if opts.Epoch == nil {
 		bklog.G(ctx).Warn("rewrite-timestamp is specified, but no source-date-epoch was found")
 		return remote, nil
 	}
+	immDiffIDs := make(map[digest.Digest]struct{})
+	if srcImg != nil {
+		for _, d := range srcImg.RootFS.DiffIDs {
```

Comment:
This needs to validate the start of the array that can be skipped, not random individual layers.

Reply:
Sorry, I don't get.
Do you mean this has to be compared by an array index, not by diffid?

Reply:
Check that the layers in the base image config are the first layers in exported config.

Eg. `diffIDs: [a, b, c]` base `diffIDs: [a, b, c, d, e]` result. First three layers would be skipped.
Eg. `diffIDs: [a, b, c]` base `diffIDs: [a, d, b, c, e]` result. No layers would be skipped.

Reply:
Fixed